docs/lab/07/index.md

```diff
@@ -0,0 +1,263 @@
+
+// I2C definition
+let mut i2c = I2c::new_async(peripherals.I2C0, scl, sda, Irqs, I2cConfig::default());
+```
+
+In section 5.2.1 and 5.2.2 of the datasheet, we get the information we need in order to read/write to a register of the BMP280 using I2C.
+
+#### Reading a register
+
+![i2c_bmp280_read](images/i2c_bmp280_read.png)
+
+To read the value of a register, we first need to send the BMP280 the address of the register we want to read. Afterwards, the sensor will send back the value of the register we requested.
+
+For this, we need to first *write* this register address over I2C and then *read* the value we get back from the sensor. We could use the `write_read_async` function to do this.
+
+```rust
+let tx_buf = [REG_ADDR]; // the tx buffer contains the address of the register we want to read
+let mut rx_buf = [0x00u8]; // the rx buffer initially contains one empty value, which will be replaced by the value of the register we asked for
+i2c.write_read_async(BMP280_ADDR, tx_buf, &mut rx_buf).await.unwrap(); // we provide the function the I2C address of the BMP280 and the two buffers
+```
+
+:::info
+I2C is *half-duplex*. The `write_read_async` function performs two separate transfers: a write and a read. As opposed to SPI, these transfers are unidirectional, and that's why we need two of them, separately.
+:::
+
+:::info
+Like with SPI, we can also read multiple registers with consecutive addresses at a time. All we need to do is modify the size of the receive buffer to be able to hold more register values, the rest of the procedure is the same.
+
+```rust
+let mut rx_buf = [0x00u8; 3];
+```
+This is explained in section 5.3 of the datasheet.
+:::
+
+#### Writing to a register
+
+![i2c_bmp280_write](images/i2c_bmp280_write.png)
+
+To write to a register, we need to send the sensor a buffer containing pairs of register addresses and values we want to write to those registers. For example, if we wanted to write `0x00` to `REG_A`:
+
+```rust
+let tx_buf = [REG_A, 0x00];
+i2c.write_async(BMP280_ADDR, tx_buf).await.unwrap();
+```
+
+If we wanted to write both `REG_A` and `REG_B` to `0x00`:
+
+```rust
+let tx_buf = [REG_A, 0x00, REG_B, 0x00];
+i2c.write_async(BMP280_ADDR, tx_buf).await.unwrap();
+```
+
+## Exercises (in progress)
+
+1. Connect the BMP280 **with I2C**. Use Kicad to draw the schematic. (**1p**)
+
+2. Read the value of the `id` register using I2C and print it over serial. (**2p**)
+
+3. Get the pressure and temperature readings from the sensor, **using I2C**. 
+- Write the `ctrl_meas` register with appropiate configuration values. You can find information on the contents you should write to this register at section 4.3.4 of the datasheet. (**2p**)
+- Read the raw pressure value and print it over the serial. (**2p**)
+- Read the raw temperature value and print it over the serial. (**1p**)
+
+:::tip
+Use what you learned in the last lab.
+:::
+
+4. Show the temperature and pressure values on the screen. (**2p**)
```

Comment:
Can we add another exercise with USB? The purpose being to register a serial device? We can start from an embassy example that echoes the text and ask the students to change the name of the device and echo back the reversed text.